File: 4차 과제/5-3/C 소스코드/5-3.c
#include <stdio.h>
#include <stdlib.h>
#include "5-3.h"

int main() {
    top=NULL;
    elment_1 expression[]="{(A+b)-3}*5+[{cos(x+y)+7}-1]*4";

    for (int i = 0;expression[i]!='\0';i++) {
        switch (expression[i]) {
        case '{':
            push(expression[i]);
            continue;
        case '}':
            pop();
            continue;
        case '(':
            push(expression[i]);
            continue;
        case ')':
            pop();
            continue;
        case '[':
            push(expression[i]);
            continue;
        case ']':
            pop();
            continue;
        }
    }
    printf("%s\n", expression);
    if (top==NULL) printf("수식의 괄호가 맞게 사용되었습니다!");
    return 0;
}

File: DataStructure/4차 과제/5-3/py 소스코드/5-3.py
from header5_3 import Stack
from func5_3_1 import LinkdedStack

def main():
    list=LinkdedStack()
    expression="{(A+B)-3}*5+[{cos(x+y)+7}-1]*4"
    print(expression)
    for i in expression:
        match i:
            case "(" | "{" | "[":
                list.push_func(i)
            case ")" | "}" | "]":
                list.pop_func()
    if list.top is None:
        print("수식의 괄호가 맞게 사용되었습니다!")
    else:
        print("수식의 괄호가 맞게 사용되지 않았습니다!")

if __name__ == "__main__":
    main()